### ISSUE TITLE
ChillerAbsorption is missing implementation of tertiary nodes for Generator Loop

### ISSUE THREAD

**Issue by jmarrec** [NREL/OpenStudio#3111]
`ChillerAbsorption` doesn't have tertiary nodes implemented to be able to connect the Generator loop. Connecting the Chilled Water and Condenser loop is now possible, but not the generator.

A very similar object, `ChillerAbsorptionIndirect` does have these connections defined (see [here](https://github.com/NREL/OpenStudio/blob/8de135fac78eda3b58b8381aac1f966e2c4c3186/openstudiocore/src/model/ChillerAbsorptionIndirect_Impl.hpp#L75)), and there is no override of the `addToNode` nor `addToTertiaryNode` methods currently.

After implementing the tertiaryInletPort and tertiaryOutletPort for `ChillerAbsorption`, in both `ChillerAbsorption` and `ChillerAbsorptionIndirect` I suggest overwriting the `addToNode` and `addToTertiaryNode` like I did in CentralHeatPumpSystem so that it'll work in the OS App. 

Basically in the addToNode, you check if the object is trying to be connected to the ***demand** side of a plant loop, and its already connected to the ***condenser** water loop but without a tertiary loop, then you connect it to the tertiary loop. See docstrings https://github.com/NREL/OpenStudio/blob/8de135fac78eda3b58b8381aac1f966e2c4c3186/openstudiocore/src/model/CentralHeatPumpSystem_Impl.hpp#L86-L94  and actual implementation [here](https://github.com/jmarrec/OpenStudio/blob/develop/openstudiocore/src/model/CentralHeatPumpSystem.cpp#L194), 
and a note I left in 

https://github.com/NREL/OpenStudio/blob/8de135fac78eda3b58b8381aac1f966e2c4c3186/openstudiocore/src/openstudio_lib/HVACSystemsController.cpp#L600-L609
